client/app/components/queries/query-editor.js

```diff
@@ -103,8 +103,10 @@ function queryEditor(QuerySnippet, $timeout) {
                 // as it makes typing slower.
                 if (tokensCount > 5000) {
                   editor.setOption('enableLiveAutocompletion', false);
+                  editor.setOption('enableBasicAutocompletion', false);
                 } else {
                   editor.setOption('enableLiveAutocompletion', true);
+                  editor.setOption('enableBasicAutocompletion', true);
```

Comment:
@jezdez to answer your question about how this makes the performance faster, the key to the performance improvement is in these 2 lines about `enableBasicAutocompletion`. The text editor redash uses is Ace Editor and without these lines there were autocompletion calls happening via the default configuration through Ace Editor that were causing a performance issue. 

Yes, a lot of other things seem to have ended up in the commit that this PR is porting upstream.

Reply:
Looking further into this, this seems to actually just add an additional command to the editor (https://github.com/ajaxorg/ace/blob/51a9b7c74671b86dceab01296a7d09e877853484/lib/ace/ext/language_tools.js#L175-L203) that triggers the auto-complete menu when typing `ctrl+space` ([et al.](https://github.com/ajaxorg/ace/blob/51a9b7c74671b86dceab01296a7d09e877853484/lib/ace/autocomplete.js#L433-L445)). Since this is *in addition* to the live auto-completion (where "live" means that is happens on any input change of the editor), I don't see how this improves performance per-se. Instead this seems like a drive-by change that was shipped in the original code change in the Mozilla fork. Not a big deal, but just want to state that this is a thing.